modules/trusted-firmware-m/CMakeLists.txt

```
@@ -183,6 +208,7 @@ function(trusted_firmware_build)
   add_library(tfm_api
     ${ZEPHYR_BASE}/modules/trusted-firmware-m/src/zephyr_tfm_log.c
   )
+  zephyr_sources(src/zephyr_tfm_psa_test.c)
```

I still think you should be using the `zephyr_sources_ifndef()` version here, until #34380 goes in.
```suggestion
  zephyr_sources_ifndef(CONFIG_TFM_PSA_TEST_NONE src/zephyr_tfm_psa_test.c)
```

---

@tejlmand Thanks, done.